Repository/LazyServiceEntityRepository.php

```diff
@@ -18,41 +20,68 @@
  */
 class LazyServiceEntityRepository extends EntityRepository implements ServiceEntityRepositoryInterface
 {
-    use LazyGhostTrait {
-        createLazyGhost as private;
-    }
+    private ManagerRegistry $registry;
+    private string $entityClass;
 
     /**
      * @param string $entityClass The class name of the entity this repository manages
      * @psalm-param class-string<T> $entityClass
      */
     public function __construct(ManagerRegistry $registry, string $entityClass)
     {
-        $initializer = function ($instance, $property) use ($registry, $entityClass) {
-            $manager = $registry->getManagerForClass($entityClass);
+        $this->registry    = $registry;
+        $this->entityClass = $entityClass;
 
-            if ($manager === null) {
-                throw new LogicException(sprintf(
-                    'Could not find the entity manager for class "%s". Check your Doctrine configuration to make sure it is configured to load this entity’s metadata.',
-                    $entityClass,
-                ));
-            }
+        if ($this instanceof LazyObjectInterface) {
+            $this->initialize();
 
-            parent::__construct($manager, $manager->getClassMetadata($entityClass));
+            return;
+        }
 
-            return $this->$property;
-        };
+        unset($this->_em);
+        unset($this->_class);
+        unset($this->_entityName);
+    }
 
-        if ($this instanceof LazyObjectInterface) {
-            $initializer($this, '_entityName');
+    /**
+     * @param string|int $name
+     *
+     * @return mixed
+     */
+    public function __get($name)
```

Comment:
```suggestion
    /** @return mixed */
    public function __get(string $name)
```

I don't understand how `$name` can be an `int`. 🤔

Reply:
nvm updated